### ISSUE TITLE
Model's _removeSupplementalFields removes fields that shouldn't be removed

### ISSUE THREAD

**Issue by alesgenova** [girder/girder#2883]
I found some unexpected behavior in one of our girder apps. (Girder 2.x, but the same is true on 3.x)

When providing to a model's `load()` method a `fields` argument of the following form, the fields are not properly included in the returned document.
```python
fields = ['foo.bar.baz']
doc = SomeModel().load(id, fields=fields)
```

While nested keys such as `'foo.bar.baz'` work just fine with mongodb projections, it seems like the model `_removeSupplementalFields` method checks for them literally in the document, and removes them:

```python
# girder/models/model_base.py

@staticmethod
def _removeSupplementalFields(doc, fields):
    # ...
    if isinstance(fields, dict):
        # ...
    else:
        for k in list(six.viewkeys(doc)):
            if k not in fields and k != '_id':
                del doc[k]
```

**Comment by zachmullen** [comment]
This is true, right now this only works on top-level fields. It would be a nice feature if it could support subfields, though; I would approve such a pull request.
